chia/wallet/util/compute_hints.py

```diff
@@ -10,22 +11,36 @@
 from chia.util.ints import uint64
 
 
-def compute_spend_hints_and_additions(cs: CoinSpend) -> Tuple[Dict[bytes32, bytes32], Dict[bytes32, Coin]]:
+@dataclass(frozen=True)
+class HintedCoin:
+    coin: Coin
+    hint: Optional[bytes32]
+
+
+def compute_spend_hints_and_additions(cs: CoinSpend) -> List[HintedCoin]:
     _, result_program = cs.puzzle_reveal.run_with_cost(INFINITE_COST, cs.solution)
 
-    hint_dict: Dict[bytes32, bytes32] = {}  # {coin_id: hint}
-    coin_dict: Dict[bytes32, Coin] = {}  # {coin_id: Coin}
+    hinted_coins: List[HintedCoin] = []
     for condition in result_program.as_iter():
         if condition.at("f").atom == ConditionOpcode.CREATE_COIN:  # It's a create coin:
             coin: Coin = Coin(cs.coin.name(), bytes32(condition.at("rf").atom), uint64(condition.at("rrf").as_int()))
-            coin_id: bytes32 = coin.name()
-            coin_dict[coin_id] = coin
+            hint: Optional[bytes32] = None
             if (
                 condition.at("rrr") != Program.to(None)  # There's more than two arguments
                 and condition.at("rrrf").atom is None  # The 3rd argument is a cons
             ):
                 potential_hint: bytes = condition.at("rrrff").atom
                 if len(potential_hint) == 32:
-                    hint_dict[coin_id] = bytes32(potential_hint)
+                    hint = bytes32(potential_hint)
+            hinted_coins.append(HintedCoin(coin, hint))
+
+    return hinted_coins
+
 
-    return hint_dict, coin_dict
+def compute_hint_for_coin(coin_name: bytes32, coin_spend: CoinSpend) -> bytes32:
+    for hinted_coin in compute_spend_hints_and_additions(coin_spend):
+        if hinted_coin.coin.name() == coin_name:
+            if hinted_coin.hint is None:
+                raise ValueError(f"No hint found for {coin_name} in {coin_spend}")
+            return hinted_coin.hint
+    raise ValueError(f"{coin_name} not found in {coin_spend}")
```

Comment:
I don't like this function existing.  It might tempt someone to run it in a loop and end up running the spend a bunch of times.

Reply:
I think the `compute_` prefix signaled this might be a heavy thing and im not sure how i feel about avoiding helpers because they could be misused which can be avoided via proper reviews also. Anyway, i dropped and in the relevant cases on the call site i think we can already assume the coin is in the spend and just access it and also we can assert the hint is there and just have a assertion message for the case there isn't? See  [b9186ed](https://github.com/Chia-Network/chia-blockchain/pull/15547/commits/b9186ed2ba3c8b6116557c065248669b1bd77afb).